### DISCUSSION TITLE
Conditional Expression Syntax

### DISCUSSION THREAD

**Discussion by jeking3** [community/community#25014] [Actions]
<p>I’m looking for the ability to use a conditional within an expression.  For example, if the github.head_ref starts with a certain prefix, I would like to change an environment variable for every job in the workflow.  (Yes, I could make two workflows, but since GitHub Actions / YAML in general doesn’t support including other YAML files, well, code duplication sucks no matter where it is…)</p>
<p>Something like:</p>
<pre><code>env:
  CONTAINER: ${{ if(!startsWith(github.head_ref, "special/"), "org/acct:latest", "org/acct:special") }}
</code></pre>
<p>My use case is that depending on the head_ref, or perhaps on the content of the pull request content, I would enable a different behavior in the workflow through an environment variable (for example choosing a different docker container to use over the default).</p>
<p>What I’d really like to see is jinja2 expressions allowed in the YAML though.  That would allow for all sorts of flexible behavior that doesn’t look like an excel function.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>For top level env, which is used for all jobs in the workflow, it doesn’t accept ‘if’ expression.  You cannot set conditional expression for it.</p>
<p>As an alternative, you can set if expression in ‘job’ level. It will exclude ‘conditional2’ jobs below:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/f4d3b7845fac33d27c08ea54c0832f695e342cba.png" data-download-href="/uploads/short-url/yVQ6s4FBZjylMpgAb48x2afUm1Q.png?dl=1" title="JobCondition.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181088426-82f708ec-8b59-40cf-bafb-7991c94803af.png" alt="JobCondition.png" data-base62-sha1="yVQ6s4FBZjylMpgAb48x2afUm1Q" width="556" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/f4d3b7845fac33d27c08ea54c0832f695e342cba_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">JobCondition.png</span><span class="informations">608×546 16.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Or you can set in step level.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/9caca88feb3a4f45a962bf0ca9fdc726bc187fe6.png" data-download-href="/uploads/short-url/mm0p4wWvWTP73OxYgxseRlrpYmq.png?dl=1" title="Stepcondition.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181088428-9084522e-b8f2-4b9a-939c-6f3133214c86.png" alt="Stepcondition.png" data-base62-sha1="mm0p4wWvWTP73OxYgxseRlrpYmq" width="580" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/9caca88feb3a4f45a962bf0ca9fdc726bc187fe6_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Stepcondition.png</span><span class="informations">628×541 15.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Regarding the jinja2 expressions, it’s recommended to raise your idea <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">here</a> where github actions manager will take a review. Thanks.</p>

**Comment by jeking3** [comment]
<p>Thanks, using ::set-env allows me to conditionally set them based on shell logic.</p>
